src/server/api/routers/rooms.ts

```diff
@@ -53,5 +53,15 @@ export const roomsRouter = createTRPCRouter({
         .where(eq(rooms.id, input.id))
         .returning();
       return res;
-    }) 
+    }),
+    show: teacherProcedure
+    .input(
+      z.object({
+        id: z.string()
+      }),
+    )
+    .query(async ({ ctx, input }) => {
+      const res = await ctx.db.select().from(rooms).where(eq(rooms.id, input.id));
```

Comment:
Looks great just add a limit and destructure the array that it returns so that this procedure returns a single room or undefined, rather than potentially multiple rooms. This is just for typesafety as we know that room ids have to be unique it should only ever return one anyway.
```suggestion
      const [res] = await ctx.db.select().from(rooms).where(eq(rooms.id, input.id)).limit(1);
```